activesupport/lib/active_support/testing/method_call_assertions.rb

```diff
@@ -35,6 +35,35 @@ def assert_not_called(object, method_name, message = nil, &block)
           assert_called(object, method_name, message, times: 0, &block)
         end
 
+        # TODO: No need to resort to #send once support for Ruby 2.4 is
+        # dropped.
+        def assert_called_on_instance_of(klass, method_name, message = nil, times: 1, returns: nil)
+          times_called = 0
+          klass.send(:define_method, "stubbed_#{method_name}") do |*|
+            times_called += 1
+
+            returns
+          end
+
+          klass.send(:alias_method, "original_#{method_name}", method_name)
+          klass.send(:alias_method, method_name, "stubbed_#{method_name}")
+
+          yield
+
+          error = "Expected #{method_name} to be called #{times} times, but was called #{times_called} times"
+          error = "#{message}.\n#{error}" if message
+
+          assert_equal times, times_called, error
+        ensure
+          klass.send(:alias_method, method_name, "original_#{method_name}")
+          klass.send(:undef_method, "original_#{method_name}")
+          klass.send(:undef_method, "stubbed_#{method_name}")
```

Comment:
Why would we want Ruby to "still search superclasses and mixed-in modules for a possible receiver" of `"original_#{method_name}"` or `"original_#{method_name}"`?

Reply:
Maybe we shouldn't since using of `undef_method` caused an issue in https://github.com/rails/rails/pull/33162#issuecomment-408616175, https://github.com/rails/rails/pull/33162#issuecomment-409002664

Reply:
I don't see that it causes any issue.